Makefile

```diff
@@ -266,11 +266,28 @@ test-sim-determinism:
 test-sim-bench:
 	@VERSION=$(VERSION) go test -benchmem -run ^BenchmarkFullAppSimulation -bench ^BenchmarkFullAppSimulation -cpuprofile cpu.out $(PACKAGES_SIM)
 
-test-e2e:
+# test-e2e runs a full e2e test suite
+# deletes any pre-existing Osmosis containers before running.
+#
+# Attempts to delete Docker resources at the end.
+# May fail to do so if stopped mid way.
+# In that case, run `make e2e-remove-resources`
+# manually.
+# Utilizes Go cache.
+test-e2e: e2e-setup
 	@VERSION=$(VERSION) OSMOSIS_E2E_UPGRADE_VERSION="v12" OSMOSIS_E2E_DEBUG_LOG=True go test -tags e2e -mod=readonly -timeout=25m -v $(PACKAGES_E2E)
 
-test-e2e-short:
-	@VERSION=$(VERSION) OSMOSIS_E2E_SKIP_UPGRADE=True OSMOSIS_E2E_SKIP_IBC=True OSMOSIS_E2E_SKIP_STATE_SYNC=True OSMOSIS_E2E_DEBUG_LOG=True go test -tags e2e -mod=readonly -timeout=25m -v $(PACKAGES_E2E)
+# test-e2e-debug runs a full e2e test suite but does
+# not attempt to delete Docker resources at the end.
+test-e2e-debug: e2e-setup
+	@VERSION=$(VERSION) OSMOSIS_E2E_UPGRADE_VERSION="v12" OSMOSIS_E2E_DEBUG_LOG=True OSMOSIS_E2E_SKIP_CLEANUP=True go test -tags e2e -mod=readonly -timeout=25m -v $(PACKAGES_E2E) -count=1
```

Comment:
Not for this PR, but now that we are defining the version in multiple places we should probably export this to a single location (and would be awesome if it could just pull the current version and add 1 to it automatically to prevent the need for manually updating this)

Reply:
Extracted a variable with the version.

I think we can't automate bumping up the version right now because it depends on the presence of the upgrade handler.

In the future, we can code gen the upgrade handler as part of the release process that would allow us to automate the version bump as well